### PR TITLE
refactor cat

### PR DESCRIPTION
**Pull request by d-513**
The cat had to be taken to the vet.

This adds an easter egg where cat is sometimes polykitty (Closes #438)
This earth-shattering change will vastly improve the PolyMC user experience.

The cat is now ephermal, and now the cat toggle isn't saved as a setting meaning the cat stays hidden and saves a few bytes

### CODE REVIEW

**Comment by Scrumplex** [comment]
Imagine lowkey removing cat smh my head

**Comment by d-513** [comment]
you mean improving

**Comment by glowiak** [comment]
mind blowing

File: launcher/ui/MainWindow.cpp [hunk 1]
```diff
@@ -914,13 +914,9 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), ui(new MainWindow
...
     // The cat background
-    {
-        bool cat_enable = APPLICATION->settings()->get("TheCat").toBool();
-        ui->actionCAT->setChecked(cat_enable);
-        // NOTE: calling the operator like that is an ugly hack to appease ancient gcc...
-        connect(ui->actionCAT.operator->(), SIGNAL(toggled(bool)), SLOT(onCatToggled(bool)));
-        setCatBackground(cat_enable);
-    }
+    // NOTE: calling the operator like that is an ugly hack to appease ancient gcc...
+    connect(ui->actionCAT.operator->(), SIGNAL(toggled(bool)), SLOT(onCatToggled(bool)));
```
Comment (kthchew) [inline launcher/ui/MainWindow.cpp]: Is a hack like this still needed for any version of gcc people will actually use today?

Comment (d-513) [reply launcher/ui/MainWindow.cpp]: probably not, I kept it in there because it was there already

Comment (d-513) [reply launcher/ui/MainWindow.cpp]: done

**Review by txtsd — APPROVED** [review]
CHAOTIC NEUTRAL

**Comment by Scrumplex** [comment]
In all seriousness, why not save the state of the cat anymore? I don't see a big issue here

**Comment by xppcom** [comment]
> In all seriousness, why not save the state of the cat anymore? I don't see a big issue here

As someone that has the cat always on , I want to keep it :(

**Comment by d-513** [comment]
Ok I thought nobody would be crazy enough to do that

**Comment by DioEgizio** [comment]
why not, cat best

**Comment by DioEgizio** [comment]
also change the cat party hats to the polymc birthday

**Comment by d-513** [comment]
> also change the cat party hats to the polymc birthday

I was gonna do it but I forgor :skull:

File: launcher/Application.cpp [hunk 2]
```diff
@@ -714,9 +714,6 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
         m_settings->registerSetting({"PreLaunchCommand", "PreLaunchCmd"}, "");
         m_settings->registerSetting({"PostExitCommand", "PostExitCmd"}, "");
 
-        // The cat
-        m_settings->registerSetting("TheCat", false);
-
```
Comment (Scrumplex) [inline launcher/Application.cpp:719]: bring back

File: launcher/ui/MainWindow.cpp [hunk 3]
```diff
@@ -1510,7 +1505,6 @@ void MainWindow::downloadUpdates(GoUpdate::Status status)
 void MainWindow::onCatToggled(bool state)
 {
     setCatBackground(state);
-    APPLICATION->settings()->set("TheCat", state);
```
Comment (Scrumplex) [inline launcher/ui/MainWindow.cpp:1513]: bring back

File: launcher/ui/MainWindow.cpp [hunk 4]
```diff
@@ -914,13 +914,8 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), ui(new MainWindow
         ui->horizontalLayout->addWidget(view);
     }
     // The cat background
-    {
-        bool cat_enable = APPLICATION->settings()->get("TheCat").toBool();
-        ui->actionCAT->setChecked(cat_enable);
```
Comment (Scrumplex) [inline launcher/ui/MainWindow.cpp:919]: bring back

**Review by Scrumplex — CHANGES_REQUESTED** [review]

**Comment by Scrumplex** [comment]
@dada513 why did you close this? :O

**Comment by d-513** [comment]
> @dada513 why did you close this? :O

Yes

**Pull request closed without merging**